modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationReturnTypeModelValidator.java

```diff
@@ -88,9 +92,16 @@ private void validateNonBlockingCallback(MethodElement<? extends Type> operation
     operationMethod.getParameters().stream()
         .filter(p -> p.getType().isSameType(CompletionCallback.class))
         .findFirst().ifPresent(p -> {
-          if (p.getType().getGenerics().isEmpty()) {
+          List<TypeGeneric> generics = p.getType().getGenerics();
+          if (generics.isEmpty()) {
             problemsReporter.addError(new Problem(p, format(MISSING_GENERICS_ERROR_MESSAGE, operationModel.getName(),
                                                             extensionModel.getName(), CompletionCallback.class.getName())));
+          } else {
+            if (generics.get(0).getConcreteType().isSameType(Void.class) &&
+                !generics.get(1).getConcreteType().isSameType(Void.class)) {
+              problemsReporter.addError(new Problem(p, format(INVALID_GENERICS_ERROR_MESSAGE, operationModel.getName(),
```

Comment:
We should debate this, but I think this should only be done if compiling.

Reply:
This breaks backwards compatibility

Reply:
Do you suggest to change these errors for warnings? Or do you see viable to fail when compiling?